pkg/platform/config.go

```diff
@@ -0,0 +1,11 @@
+package platform
+
+// Client configuration provides the logic for configuring and creating a client to be used to perform platform specific operations
+type ClientConfiguration interface {
+
+	// CreateFunc creates a new platform client that will be used to communicate with the platform
+	CreateFunc() (Client, error)
```

Comment:
why don't you pass the `Client` directly? 
you call this `CreateFunc` only once in `defaultRegJob`

Reply:
I wanted to be consistent with the other clients creation  - one of them (osb) requires creation of a new client on each request - so i wanted all the clients to be set up in the same way.